merge read_bin_tiles functions

timothyas opened the pull request:
This addresses the read_bin_llc functions which were added with #42. Rather than raising an issue, I figured I would just make the PR and we can talk about it here. I propose two changes: 

1. Rather than maintain two separate functions for reading llc tiles (the "dirty" way, rather than using `load_ecco_vars_from_mds` which I think is preferred), one which uses the xmitgcm function and one which does not, I propose to just have one `read_llc_to_tiles` with the additional option `use_xmitgcm`. There are already quite a lot of functions for reading binaries, and in my opinion it seems a bit cleaner and easier to maintain. 

2. I think the default should still be to use the xmitgcm function because it "does less" and it by default reads functions in as a dask array. Now that the dimension order is sorted out and there is no more axis swapping (#45), arrays are simply "read in". The other option is to read in arrays in compact form, which loads the array into memory, and then converts this array to faces format, then to tiles format. The former seems more streamlined. 

@ifenty I'm curious if you agree since earlier you mentioned that using the xmitgcm function was causing a lot of overhead, but reading into compact form then converting to faces then tiles seems like it is more overhead. Let me know what you think!

ifenty commented:
@timothyas I like the idea of merging and having a use_xmitgcm option.  

One problem was that there are many files written in the compact format that are not generated by the model (they are mainly  *inputs* to the model).  I could be wrong but I thought that xmitgcm requires *.data/*.meta file pairs.  Because of this it isn't obvious to me that the use_xmitgcm option should be true by default.  Another issue is that sometimes the model does have *.meta* files but the output isn't a diagnostic.  I'm thinking in particular about model fields written to support the cost calculations.  they have .meta files but include many time levels in a single file (the so-called bar files).  @owang01  can comment but I don't believe that xmitgcm can handle those files gracefully.

I don't recall which steps xmitgcm goes through from when it reads the binary file in compact format until its spits out the fields as tiles, but according to @owang01 the xmitgcm read step was tons slower in at least one case.  Maybe it was a case with many small 2D fields that easily fit into memory and loading them with xmitgcm took a long time to establish the dask structures that were immediately tossed out anyway after *.load* or *.compute*.

owang01 commented:
I also like the idea of merging and having a use_xmitgcm option. 

I think the use_xmitgcm option should be false by default. The mean reason is that using xmitgcm seems much slower (<40%) than otherwise. Here is the benchmark result of read_llc_to_tiles and read_llc_to_tiles_xmitgcm reading an m_bpmon file with 288 2D llc90 fields:
read_llc_to_tiles: 8.5 seconds 
read_llc_to_tiles_xmitgcm: 11.7 seconds

timothyas commented:
Huh, that's interesting! Well, that's convincing enough to me especially since this. I changed the default to `use_xmitgcm=False`.

Also, I noticed an inconsistency regarding some cases with `skip>1` and fixed those as well. Now the two functions are identical aside from the dask stuff...
